integration-tests/pkg/mock_sensor/server.go

```diff
@@ -489,9 +489,33 @@ func (m *MockSensor) pushEndpoint(containerID string, endpoint *sensorAPI.Networ
 // translateAddress is a helper function for converting binary representations
 // of network addresses (in the signals) to usable forms for testing
 func (m *MockSensor) translateAddress(addr *sensorAPI.NetworkAddress) string {
+	address := utils.IPAddress{}
+	ipNetwork := utils.IPNetwork{}
+
+	addressData := addr.GetAddressData()
+	if len(addressData) > 0 {
+		address = utils.IPFromBytes(addressData)
+	} else {
+		// If there is no address data IpNetwork should be set and represent
+		// a CIDR block or external IP address.
+		ipNetworkData := addr.GetIpNetwork()
+		if len(ipNetworkData) > 0 {
+			ipNetwork = utils.IPNetworkFromCIDRBytes(ipNetworkData)
+			// If the prefix length is 32 this is a regular IP address
+			// and not a CIDR block
+			if ipNetwork.PrefixLen() == byte(32) {
+				address = ipNetwork.IP()
+				ipNetwork = utils.IPNetwork{}
+			}
+		}
+
+	}
+
 	ipPortPair := utils.NetworkPeerID{
-		Address: utils.IPFromBytes(addr.GetAddressData()),
-		Port:    uint16(addr.GetPort()),
+		Address:   address,
+		IPNetwork: ipNetwork,
+		Port:      uint16(addr.GetPort()),
 	}
+
 	return ipPortPair.String()
 }
```

Comment:
This would be more readable if you use early returns, something like this:
```go
func (m *MockSensor) translateAddress(addr *sensorAPI.NetworkAddress) string {
	peerId := utils.NetworkPeerID{Port: uint16(addr.GetPort())}
	addressData := addr.GetAddressData()
	if len(addressData) > 0 {
		peerId.Address = utils.IPFromBytes(addressData)
		return peerId.String()
	}

	// If there is no address data, IpNetwork should be set and represent
	// a CIDR block or external IP address.
	ipNetworkData := addr.GetIpNetwork()
	if len(ipNetworkData) == 0 {
		return peerId.String()
	}

	// If the prefix length is 32 this is a regular IP address
	// and not a CIDR block
	ipNetwork := utils.IPNetworkFromCIDRBytes(ipNetworkData)
	if ipNetwork.PrefixLen() == byte(32) {
		peerId.Address = ipNetwork.IP()
	} else {
		peerId.IPNetwork = ipNetwork
	}
	return peerId.String()
}
```

Reply:
Done